main.c

```diff
@@ -333,7 +342,13 @@ static void *dofastwork(void *task)
 #ifdef STATISTICS
 	++st->numrestart.v;
 #endif
-	randombytes(seed,sizeof(seed));
+	pthread_mutex_lock(&globalseed_mutex);
```

Comment:
code duplication, could easily be avoided with additional static function

Reply:
More of the same.